msgpack-core/src/test/scala/org/msgpack/core/MessageUnpackerTest.scala

```diff
@@ -663,5 +664,30 @@ class MessageUnpackerTest extends MessagePackSpec {
         unpacker.getTotalReadBytes shouldBe arr.length
       }
     }
+
+    "unpack string crossing end of buffer" in {
+      def check(expected: String, strLen: Int) = {
+        val bytes = new Array[Byte](strLen)
+        val out = new ByteArrayOutputStream
+
+        val packer = MessagePack.newDefaultPacker(out)
+        packer.packBinaryHeader(bytes.length)
+        packer.writePayload(bytes)
+        packer.packString(expected)
+        packer.close
+
+        val unpacker = new MessageUnpacker(new InputStreamBufferInput(new ByteArrayInputStream(out.toByteArray)))
+        val len = unpacker.unpackBinaryHeader
+        unpacker.readPayload(len)
+        val got = unpacker.unpackString
+        unpacker.close
+
+        got shouldBe expected
+      }
+
+      Seq("あ", "aあ", "あa", "あいうえおかきくけこさしすせそ").foreach { s =>
```

Comment:
Could you replace this into \uXXXX based strings, to avoid using Japanese characters within the code?